eng/pipelines/source-build-license-scan.yml

```diff
@@ -0,0 +1,124 @@
+schedules:
+- cron: "0 7 * * 1-5"
+  displayName: Run on weekdays at 7am UTC
+  branches:
+    include:
+    - main
+    - release/*
+
+pr: none
+trigger: none
+
+variables:
+  installerRoot: '$(Build.SourcesDirectory)/src/installer'
+
+jobs:
+- job: Setup
+  pool:
+    name: NetCore1ESPool-Svc-Internal
+    demands: ImageOverride -equals 1es-ubuntu-2004
+  steps:
+  - script: |
+      vmrSrcDir="$(Build.SourcesDirectory)/src"
+      
+      # Builds an Azure DevOps matrix definition. Each entry in the matrix is a path,
+      # allowing a job to be run for each src repo.
+      matrix=""
+      for dir in $vmrSrcDir/*/
+      do
+        if [ ! -z "$matrix" ]; then
+          matrix="$matrix,"
+        fi
+        repoName=$(basename $dir)
+        #matrix="$matrix{ \"$repoName\": { \"repoPath\": \"$dir\" } }"
+        matrix="$matrix \"$repoName\": { \"repoPath\": \"$dir\" }"
+      done
+
+      matrix="{ $matrix }"
+
+      echo "##vso[task.setvariable variable=matrix;isOutput=true]$matrix"
+    name: GetMatrix
+    displayName: Get Matrix
+
+- job: LicenseScan
+  dependsOn: Setup
+  pool:
+    name: NetCore1ESPool-Svc-Internal
+    demands: ImageOverride -equals 1es-ubuntu-2004
+  timeoutInMinutes: 420
+  strategy:
+    matrix: $[ dependencies.Setup.outputs['GetMatrix.matrix'] ]
+  steps:
+
+  - task: UseDotNet@2
+    displayName: Install .NET
+    inputs:
+      packageType: 'sdk'
+      version: '8.0.x'
+      includePreviewVersions: true
+
+  - task: PipAuthenticate@1
+    displayName: 'Pip Authenticate'
+    inputs: 
+      artifactFeeds: public/dotnet-public-pypi
+      onlyAddExtraIndex: false
+
+  - script: $(installerRoot)/eng/install-scancode.sh
+    displayName: Install Scancode
+
+  - script: >
+      dotnet test
+      $(Build.SourcesDirectory)/test/Microsoft.DotNet.SourceBuild.SmokeTests/Microsoft.DotNet.SourceBuild.SmokeTests.csproj
+      --filter "Category=LicenseScan"
```

Comment:
Does that mean running smoke tests via the command line (without any filtering) will run a full license scan too?

Reply:
No, that's not the intent currently. It needs to be run on a per-repo basis. The implementation _could_ be changed to support a full scan in one invocation but that logic would never be validated via the CI since it would always split the repos into separate jobs. Let me know your thoughts.

Reply:
The current state is good. Running a 6-hour long license scan when a user tries to run smoke test is undesired.